ledger/participant-integration-api/src/main/scala/platform/apiserver/configuration/LedgerConfigProvider.scala

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.apiserver.configuration
+
+import akka.stream.Materializer
+import com.daml.api.util.TimeProvider
+import com.daml.ledger.api.SubmissionIdGenerator
+import com.daml.ledger.participant.state.index.v2.IndexConfigManagementService
+import com.daml.ledger.participant.state.{v2 => state}
+import com.daml.ledger.resources.ResourceOwner
+import com.daml.logging.LoggingContext
+import com.daml.platform.configuration.LedgerConfiguration
+
+object LedgerConfigProvider {
+  def owner(
+      index: IndexConfigManagementService,
+      optWriteService: Option[state.WriteConfigService],
+      timeProvider: TimeProvider,
+      ledgerConfiguration: LedgerConfiguration,
+  )(implicit
+      materializer: Materializer,
+      loggingContext: LoggingContext,
+  ): ResourceOwner[CurrentLedgerConfiguration] =
+    for {
+      currentLedgerConfiguration <-
+        IndexStreamingCurrentLedgerConfiguration.owner(index, ledgerConfiguration)
+      _ <- optWriteService match {
+        case None => ResourceOwner.unit
+        case Some(writeService) =>
+          LedgerConfigProvisioner.owner(
+            currentLedgerConfiguration = currentLedgerConfiguration,
+            writeService = writeService,
+            timeProvider = timeProvider,
+            submissionIdGenerator = SubmissionIdGenerator.Random,
+            ledgerConfiguration = ledgerConfiguration,
+          )
+      }
+      _ <- ResourceOwner.forFuture(() => currentLedgerConfiguration.ready)
+    } yield currentLedgerConfiguration
```

Comment:
This still tricks me but I understand it's idiomatic. If you can think of some meaningful comment to clarify that a provisioner is hooked in to preserve the current behavior, please do add it.

Reply:
[I added some comments.](https://github.com/digital-asset/daml/pull/10455/commits/7e4567a37f68d2f0ea81da302e14776ab9498e3f)